app/src/main/java/com/support/android/designlibdemo/CheeseDetailActivity.java

```diff
@@ -42,6 +43,17 @@ public void onCreate(Bundle savedInstanceState) {
         final String cheeseName = intent.getStringExtra(EXTRA_NAME);
 
         final Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
+        if(Build.VERSION.SDK_INT>=21){
+            int result = 0;
+            int resourceId = getResources().getIdentifier(
+                    "status_bar_height", "dimen", "android");
+            if (resourceId > 0) {
+                result = getResources().getDimensionPixelSize(resourceId);
```

Comment:
Sorry, I should have explained this better. If you make the `Build.VERSION.SDK_INT == 21` alteration above, you also need to remove `android:fitsSystemWindows="true"` below as it causes the Toolbar to be positioned incorrectly on API level 22. Once those changes are made, you also need to change the above line to `result = getResources().getDimensionPixelSize(resourceId) * 2;` to position the Toolbar correctly on API level 21.